File: 13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> map = new HashMap<>();
        Map<String, Integer> map2 = new HashMap<>();
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);

        map2.put("IV", 4);
        map2.put("IX", 9);
        map2.put("XL", 40);
        map2.put("XC", 90);
        map2.put("CD", 400);
        map2.put("CM", 900);
        int sum = 0;
        int i=s.length()-1;
        for(; i>0; i--) {
            String str = s.charAt(i-1) + "" +s.charAt(i);
            System.out.println("str :"+str);

            if(map2.containsKey(str)) {
                System.out.println(map2.get(str));
                sum += map2.get(str);
                i = i - 1;
            } else if(map.containsKey(s.charAt(i))) {
                System.out.println(map.get(s.charAt(i)));
                sum += map.get(s.charAt(i));
            }
        }

        if(i == 0) {
            System.out.println(map.get(s.charAt(i)));
            sum += map.get(s.charAt(i));
        }

        return sum;
    }
}

File: 13-roman-to-integer/roman-to-integer.cs
public class Solution {
    public int RomanToInt(string s) {
        Dictionary<char, int> symb1 = new Dictionary<char, int>();
        Dictionary<string, int> symb2 = new Dictionary<string, int>();

        symb1['I'] = 1;
        symb1['V'] = 5;
        symb1['X'] = 10;
        symb1['L'] = 50;
        symb1['C'] = 100;
        symb1['D'] = 500;
        symb1['M'] = 1000;

        symb2["IV"] = 4;
        symb2["IX"] = 9;
        symb2["XL"] = 40;
        symb2["XC"] = 90;
        symb2["CD"] = 400;
        symb2["CM"] = 900;

        int sum = 0;
        int i = s.Length - 1;

        for(; i>0; i--) {
            string str = s[i-1] + "" + s[i];

            if(symb2.ContainsKey(str)) {
                sum += symb2[str];
                i = i-1;
            } else {
                sum += symb1[s[i]];
            }
        }

        if(i==0) {
            sum += symb1[s[i]];
        }

        return sum;
    }
}